### PR TITLE
Add action to deploy the simple dashboard to app.visivo.io on every pull request.

### PR DESCRIPTION
**Pull request by tgsoverly**
End to end example on how to build and deploy a dashboard on every pull request.

### CODE REVIEW

File: visivo/commands/deploy.py [hunk 1]
```diff
@@ -29,10 +29,13 @@ def deploy(working_dir, user_dir, output_dir, stage, host):
     if profile_file:
         profile = load_yaml_file(profile_file)
 
-    if not profile or "token" not in profile:
-        raise click.ClickException(
-            f"{PROFILE_FILE_NAME} not present or token not present in {PROFILE_FILE_NAME}: {user_dir}"
-        )
+    profile_token = os.getenv('VISIVO_TOKEN')
```
Comment (tgsoverly) [inline visivo/commands/deploy.py:32]: TODO: Add test

File: visivo/command_line.py [hunk 2]
```diff
@@ -39,6 +41,7 @@ def safe_visivo():
             raise e
         click.echo("An unexpected error has occurred")
         click.echo(e)
+        exit(1)
```
Comment (tgsoverly) [inline visivo/command_line.py:44]: Wasn't actually failing because of this...

Comment (jpjsonic) [reply visivo/command_line.py:44]: Ahhh, got it. Nice find

File: .github/workflows/deploy_dashoard.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          python-version: '3.10' 
+
+      - name: Install CLI 
+        run: python -m pip install git+https://github.com/visivo-io/visivo.git@${{ github.head_ref }} --force-reinstall
+
+      - name: Deploy
+        if: github.event.pull_request.closed_at == null
+        run: cd test-projects/simple-database && visivo run && VISIVO_TOKEN=${{ secrets.VISIVO_TOKEN }} visivo deploy -s ${{ github.head_ref }}
+
+      - name: Archive
```
Comment (jpjsonic) [inline .github/workflows/deploy_dashoard.yml:25]: This is great. So we can probably make this action available to people by just doing the 

```
steps:
      - name: Checkout
        uses: actions/deploy-visivo-pr-stage
```

File: visivo/commands/deploy.py [hunk 4]
```diff
@@ -77,11 +68,11 @@ def deploy(working_dir, user_dir, output_dir, stage, host):
             body = {
                 "name": trace.name,
                 "project_id": project_id,
-                "data_file_id": response.model_dump_json()["id"],
```
Comment (jpjsonic) [inline visivo/commands/deploy.py:80]: Ah so this is what was breaking deploy some pydantic 1.0 syntax

File: visivo/commands/utils.py [hunk 5]
```diff
@@ -40,6 +44,26 @@ def find_or_create_target(project: Project, target_or_name: str) -> Target:
...
+    if profile_token:
+        return profile_token
+
+    profile = None
+    if profile_file:
+        profile = load_yaml_file(profile_file)
+
+    if not profile or "token" not in profile:
+        raise click.ClickException(
+            f"{PROFILE_FILE_NAME} not present or token not present in {PROFILE_FILE_NAME}"
```
Comment (jpjsonic) [inline visivo/commands/utils.py:58]: Nice, this is a great error 🥳

**Review by jpjsonic — APPROVED** [review]
Looks great! Excited to get this one merged in. I added the secrets so we should be set there.

File: .github/workflows/deploy_dashoard.yml (hunk 3 above)
Comment (tgsoverly) [reply .github/workflows/deploy_dashoard.yml:25]: Yea, pulling it out to an action is a great idea.

File: visivo/commands/deploy.py (hunk 4 above)
Comment (tgsoverly) [reply visivo/commands/deploy.py:80]: I think it was a bad replace.

Comment (jpjsonic) [reply visivo/commands/deploy.py:80]: I agree, model dump json was a lot more clear than just json

**Pull request merged**